Sources/LanguageServerProtocol/SupportTypes/ServerCapabilities.swift

```diff
@@ -429,9 +429,27 @@ public struct CompletionItemOptions: Codable, Hashable {
   ) {
     self.labelDetailsSupport = labelDetailsSupport
   }
+
+  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+    guard case .bool(let labelDetailsSupport) = dictionary["labelDetailsSupport"] else {
+      return nil
```

Comment:
I think we should just set `labelDetailsSupport` to `nil` here instead of returning `nil` so that encoding + decoding round-trips.

Reply:
or to `false` 🤔  the initializer default to `false`, not `nil`

Reply:
Oh, interesting that the initializer defaults to `false`. At least for decoding I think `nil` is the correct value though.